standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```
@@ -2835,17 +2834,25 @@ private boolean checkTableDataShouldBeDeleted(Table tbl, boolean deleteData) {
      * @param tablePath
      * @param ifPurge completely purge the table (skipping trash) while removing
      *                data from warehouse
-     * @param db database the table belongs to
+     * @param shouldEnableCm If cm should be enabled
      */
-    private void deleteTableData(Path tablePath, boolean ifPurge, Database db) {
+    private void deleteTableData(Path tablePath, boolean ifPurge, boolean shouldEnableCm) throws MetaException {
+      if (tablePath != null) {
+        wh.deleteDir(tablePath, true, ifPurge, shouldEnableCm);
+      }
+    }
 
+    /**
+     * Deletes the data in a table's location, if it fails logs an error.
+     *
+     * @param tablePath
+     * @param ifPurge completely purge the table (skipping trash) while removing
+     *                data from warehouse
+     * @param db Database
+     */
+    private void deleteTableData(Path tablePath, boolean ifPurge, Database db) throws MetaException {
       if (tablePath != null) {
-        try {
-          wh.deleteDir(tablePath, true, ifPurge, db);
-        } catch (Exception e) {
-          LOG.error("Failed to delete table directory: " + tablePath +
-              " " + e.getMessage());
-        }
+        wh.deleteDir(tablePath, true, ifPurge, db);
```

for now ..just log the error ..ignore the exception while deleting.